browser/src/canvas/sections/URLPopUpSection.ts

```diff
@@ -111,18 +116,47 @@ class URLPopUpSection extends HTMLObjectSection {
 
 		document.getElementById(this.removeButtonId).onclick = () => {
 			app.map.sendUnoCommand('.uno:RemoveHyperlink', params);
+			URLPopUpSection.closeURLPopUp();
 		};
 	}
 
+	reLocateArrow(arrowAtTop: boolean) {
+		if (arrowAtTop) this.arrowDiv.classList.add('reverse');
+		else this.arrowDiv.classList.remove('reverse');
+
+		/*
+			Normally, the documentPosition sent from the core side nicely positions the arrow.
+			We will check if we reposition the popup.
+			If the arrow position falls outside of the popup, we will put it on the edge.
+		*/
+		const clientRect = this.getPopUpBoundingRectangle();
+		const arrowHalfWidth = 10;
+		let arrowCSSLeft = this.sectionProperties.documentPosition.pX - this.documentTopLeft[0] + this.containerObject.getDocumentAnchor()[0] - arrowHalfWidth;
+		arrowCSSLeft /= app.dpiScale;
+		arrowCSSLeft += document.getElementById('canvas-container').getBoundingClientRect().left; // Add this in case there is something on its left.
+		arrowCSSLeft -= clientRect.left;
+		this.arrowDiv.style.left = (arrowCSSLeft > 6 ? arrowCSSLeft : 6) + 'px'; // 6: padding of the popup.
+	}
+
 	public static resetPosition(section: URLPopUpSection) {
 		if (!section)
 			section = app.sectionContainer.getSectionWithName(URLPopUpSection.sectionName);
 
-		const left = section.sectionProperties.documentPosition.pX - section.getPopUpWidth() * 0.5 * app.dpiScale;
-		const top = section.sectionProperties.documentPosition.pY - (section.getPopUpHeight() + 20) * app.dpiScale;
+		let left = section.sectionProperties.documentPosition.pX - section.getPopUpWidth() * 0.5 * app.dpiScale;
+		let top = section.sectionProperties.documentPosition.pY - (section.getPopUpHeight() + 20) * app.dpiScale;
 
 		if (section) {
-			section.setPosition((left >= 0 ? left : 10), (top >= 0 ? top : 10));
+			let arrowAtTop = false;
+			if (top < 0) {
+				top = section.sectionProperties.documentPosition.pY + (40 * app.dpiScale);
```

Comment:
I would put all the magic numbers in single place here as memebers of a class and reuse everywhere:
```
const popupHorizontalPagging = 6;
const popupVerticalPadding = 20;
```
then we have single value to change all of them in case designers will decide that